backend/src/main/kotlin/org/loculus/backend/service/submission/SubmissionDatabaseService.kt

```diff
@@ -466,12 +451,13 @@ open class SubmissionDatabaseService(
             accessionPreconditionValidator.validate {
                 thatAccessionVersionsExist(accessionVersionsFilter)
                     .andThatOrganismIs(organism)
-                    .andThatSequenceEntriesAreInStates(listOf(Status.AWAITING_APPROVAL))
+                    .andThatSequenceEntriesAreInStates(listOf(Status.PROCESSED))
+                    .andThatSequenceEntriesHaveNoErrors()
```

Comment:
This is a key change that we should test well - is this covered by a unit test? That if you try to approve a sequence with errors an error is thrown? It probably is covered, just double checking.

Reply:
It actually wasn't covered. I agree that it's important to test. I added a test here:

[d5db67c](https://github.com/loculus-project/loculus/pull/3125/commits/d5db67c4b4cdcad892a6bbf716c5299e3074a746)